src/dosbox.cpp

```diff
@@ -584,6 +583,13 @@ void DOSBOX_Init(void) {
 	                  "scan2x, scan3x, tv2x, tv3x, sharp (default).");
 #endif
 
+	pstring = secprop->Add_path("maxresolution", always, "auto");
+	pstring->Set_help(
```

Comment:
When I wrote my new config file (`Z:\> config -wc`), I scrolled down and was looking for the `maxresolution` setting along side the other `resolutions, but couldn't find it:

![2021-09-22_06-30](https://user-images.githubusercontent.com/1557255/134368999-5ddb3b5f-6bdf-4cdb-b792-4c20ae997756.png)

Instead, I had to scroll down a couple pages into the `[render]` section:

![2021-09-22_06-31](https://user-images.githubusercontent.com/1557255/134368997-868cf79b-c041-4f42-9a0d-d33c6c170208.png)

If a user's going to be fine-tuning the sizing of their fullscreen mode, then they will be adjusting these two in tandem - so makes sense to pair them in in the conf file (given both involve content sizing as well).

Placement? I'd say, because maxresolution applies to both `fullscreenresolution` and `windowresolution`, it probably makes sense to put it above `fullscreenresolution`.  Although it's a setting that most users probably will leave as auto - so making it come after these two, just above  `vsync`, might be even better.  Just thinking out loud.

Reply:
I've given this a lot of thought because first I wanted to put it into the `sdl` section for the same reason you wrote. But then I figured options related to the window itself (or screen size) are in the `sdl` section, and anything related to the actual rendered content lives in `render`.

Having said that, all this breaks down when you consider that `aspect`, `windowresolution`, `maxresolution` are all interdependent... **For the time being, I agree that it's best to put it after `vsync_skip`.** But the whole `sdl` section is quite problematic for several reasons:

- the name `sdl` is meaningless to the user; it's a case of implementation details leaking into the user config (Why should the user know, or even care that DOSBox uses SDL under the hood? He's not configuring "SDL"; he's configuring render options, input options, general options, etc.)
- it's pretty much a dumping ground for all sorts of different options with not much rhyme or reason

I assume you guys just kept it that way for backward-compatibility. But at some point I think it would be better to make a breaking change, rip the whole `sdl` section apart, and move individual options into logical sections related to the functionality they control. One possible solution (I just spent like 5 minutes on this, probably could be made a lot better):

```
[input]  (new section)
mapperfile
capture_mouse
sensitivity
raw_mouse_input

[dosbox]
priority
screensaver
waitonerror

[render]
fullscreen
display
fullresolution
windowresolution
windowposition
windowdecorations
vsync
vsync_skip
maxresolution
```

Reply:
I just realised that new config options use snake_case (e.g. `raw_mouse_input`), while options inherited from DOSBox don't (e.g. `windowresolution`). I much prefer the new style, so I think it's best to use that for all of my new config options:

`window_position`
`window_decorations`
`max_resolution`

That will make them inconsistent with the old DOSBox style, but I assume the goal is to eventually normalise all config options to the snake_case standard (and hopefully refactor options like 'aspect' to something sensible, e.g. `aspect_ratio_correction` or `square_pixels`).

I will keep these changes as a separate commit at the end so I can roll it back in case you really don't like the idea.

Reply:
Looks even better! Thanks @johnnovak 👍